### PR TITLE
:sparkles: create champion entity

### PR DESCRIPTION
**Pull request by MarcosOliveir4**

### CODE REVIEW

File: src/app/domain/entities/champion/champion-entity.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    expect(sut().tags).toEqual(['Fighter', 'Tank']);
+  });
+
+  it('should validate stats', () => {
+    const stats = {
+      hp: 650,
+      armor: 38,
+      attackdamage: 60,
+      attackspeed: 0.651,
+    };
```
Comment (iurynogueira) [inline src/app/domain/entities/champion/champion-entity.spec.ts:40]: Move this obj to describe outside and use in sut too.

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Comment by iurynogueira** [comment]
CI not run

**Comment by iurynogueira** [comment]
> CI not run

fixed!

Build forked pull requests
Run builds for pull requests from forks. CircleCI will automatically update the commit status shown on GitHub's pull request page.

File: src/app/domain/entities/champion/champion-entity.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+export class ChampionEntity extends DomainEntity {
+  public readonly name!: string;
+  public readonly tags!: string[];
+  public readonly stats!: {
+    hp: number;
+    armor: number;
+    attackdamage: number;
+    attackspeed: number;
+  };
```
Comment (mizek1) [inline src/app/domain/entities/champion/champion-entity.ts:22]: I think you should create an interface for this type.

File: src/app/domain/entities/champion/champion-entity.spec.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    key: '266',
+    tags: ['Fighter', 'Tank'],
+    stats: {
+      hp: 650,
+      armor: 38,
+      attackdamage: 60,
+      attackspeed: 0.651,
+    },
+  });
+};
```
Comment (mizek1) [inline src/app/domain/entities/champion/champion-entity.spec.ts:15]: I think your sut should receive an object, not mock directly in the constructor.
You can use this object to check the equalities of the attributes, and if you change a value from object, you won't need to change the test case.

File: src/app/domain/entities/champion/champion-entity.ts (hunk 2 above)
Comment (MarcosOliveir4) [reply src/app/domain/entities/champion/champion-entity.ts:22]: Sim foi criado, porem quando eu inicio essa variável o typescript força eu declara cada um dos atributos

File: src/app/domain/entities/champion/champion-entity.spec.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { ChampionEntity } from './champion-entity';
+
+const champion = {
```
Comment (iurynogueira) [inline src/app/domain/entities/champion/champion-entity.spec.ts:3]: Create mock file and export this guys.

File: src/app/domain/entities/champion/champion-entity.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+export interface ChampionProps {
+  name: string;
+  key: string;
+  tags: string[];
+  stats: {
+    hp: number;
+    armor: number;
+    attackdamage: number;
+    attackspeed: number;
+  };
```
Comment (iurynogueira) [inline src/app/domain/entities/champion/champion-entity.ts:12]: Create interface. StatsChampion {...}

**Review by iurynogueira — CHANGES_REQUESTED** [review]
Nice work!

**Pull request merged**